app/src/main/java/fr/free/nrw/commons/WelcomeActivity.java

```diff
@@ -86,7 +86,11 @@ public void onBackPressed() {
         if (pager.getCurrentItem() != 0) {
             pager.setCurrentItem(pager.getCurrentItem() - 1, true);
         } else {
-            finish();
+            if(defaultKvStore.getBoolean("firstrun",true)){
+                finishAffinity();
```

Comment:
does `finishTask` work too? I never want to have to think about task affinity

Reply:
On app resume, loginActivity keeps calling WelcomeActivity if firstrun is set true. So it's kinda stuck in loop and that's why finish() doesn't work. 
finishAndRemoveTask?

Reply:
`finishTask()`/`finishAndRemoveTask()` whatever works and doesn't mention affinities

Reply:
it requires min api 21 so i am using moveTaskToBack() instead

Reply:
hmmm that moves us even further from what we want to express. I guess `finishAffinity`  is best here, it at least has the word finish in it

Reply:
Exactly, or there is one more solution creating new intent and setting flag  FLAG_ACTIVITY_CLEAR_TOP

Reply:
Or Welcome could be started for result, a cancel could be read as a finish for login, 
pros: very explicit
cons: way too much code.

Lets keep finishAffinity. it is easy to get rid of if minds change later.